pkg/tfbridge/schema.go

```diff
@@ -365,6 +365,14 @@ func MakeTerraformInput(res *PulumiResource, name string,
 		v = nv
 	}
 
+	// If we want to mark the property as Computed then we need to do so before we process the value
+	// this will override any specific usage of Optional. This is useful in scenarios where the data source
+	// has marked the parameter as Optional + Computed but they actually just mean that it's computed
+	// e.g. tags https://github.com/pulumi/pulumi-aws/issues/752
+	if ps != nil && ps.MarkAsComputed != nil && *ps.MarkAsComputed {
```

Comment:
Do we actually need this? Seems like we should never get here...

Reply:
I will happily remove the `&& *ps.MarkAsComputed` - it was purely there to check to make sure that someone didn't make this false - we can assume that if it's there, it will always be true :)

Reply:
Sorry, I meant that I don't think we need this condition at all. It might even be incorrect, as we do call `MakeTerraformInputs` with values from the state, where these fields may be populated.

Reply:
Oh interesting, the intent of this check was to say:

1. Is this a value that we are trying to override in the schema to be Computed
2. If so, then actually mark the property value as Computed

Do you not think this is the way to do this?

Reply:
Oh interesting, the intent of this check was to say:

1. Is this a value that we are trying to override in the schema to be Computed
2. If so, then actually mark the property value as Computed

Do you not think this is the way to do this?

Reply:
No, I don't think we need this here. We really only use `Computed` in the schema to guide `tfgen`'s code generation w.r.t. inputs/state.

Reply:
Ah interesting - so the only work that this needs outside of a change to `info.go` is that we make the changes in tfgen to pick up that it's no longer an optional property but now only a computed property?

If so, can you point me in the direction of where that would live in tfgen?